web/packages/api/src/toEthereum_v3.ts

```diff
@@ -0,0 +1,856 @@
+            })
+        }
+    } else {
+        if (amount > tokenBalance) {
+            logs.push({
+                kind: ValidationKind.Error,
+                reason: ValidationReason.InsufficientTokenBalance,
+                message: "Insufficient token balance to submit transaction.",
+            })
+        }
+    }
+
+    let etherBalance = await sourceParachainImpl.getTokenBalance(
+        sourceAccountHex,
+        registry.ethChainId,
+        ETHER_TOKEN_ADDRESS
+    )
+
+    if (fee.ethereumExecutionFee! > etherBalance) {
+        logs.push({
+            kind: ValidationKind.Error,
+            reason: ValidationReason.InsufficientEtherBalance,
+            message: "Insufficient ether balance to submit transaction.",
+        })
+    }
+
+    let sourceDryRunError
+    let assetHubDryRunError
+    if (source.features.hasDryRunApi) {
+        // do the dry run, get the forwarded xcm and dry run that
+        const dryRunSource = await dryRunOnSourceParachain(
+            sourceParachain,
+            registry.assetHubParaId,
+            registry.bridgeHubParaId,
+            transfer.tx,
+            sourceAccountHex
+        )
+        if (!dryRunSource.success) {
+            logs.push({
+                kind: ValidationKind.Error,
+                reason: ValidationReason.DryRunFailed,
+                message: "Dry run call on source failed.",
+            })
+            sourceDryRunError = dryRunSource.error
+        }
+
+        if (dryRunSource.success && sourceParaId !== registry.assetHubParaId) {
+            if (!dryRunSource.assetHubForwarded) {
+                logs.push({
+                    kind: ValidationKind.Error,
+                    reason: ValidationReason.DryRunFailed,
+                    message: "Dry run call did not provide a forwared xcm.",
+                })
+            } else {
+                const dryRunResultAssetHub = await dryRunAssetHub(
+                    assetHub,
+                    sourceParaId,
+                    registry.bridgeHubParaId,
+                    dryRunSource.assetHubForwarded[1][0]
+                )
+                if (!dryRunResultAssetHub.success) {
+                    logs.push({
+                        kind: ValidationKind.Error,
+                        reason: ValidationReason.DryRunFailed,
+                        message: "Dry run failed on Asset Hub.",
+                    })
+                    assetHubDryRunError = dryRunResultAssetHub.errorMessage
+                }
+            }
+        }
+    } else {
+        logs.push({
+            kind: ValidationKind.Warning,
+            reason: ValidationReason.DryRunApiNotAvailable,
+            message: "Source parachain can not dry run call. Cannot verify success.",
+        })
+        if (sourceParaId !== registry.assetHubParaId) {
+            let dryRunResultAssetHub: any
+            if (sourceAssetMetadata.location) {
```

Comment:
Some comments to explain the different cases here, and on line 509 and 531 would be great. Or make it readable, like `ifFeeTokenIsNotDot()`.

Reply:
Basically, it's to distinguish PNA from ENA, since `assetMetadata` of PNA includes a location field, whereas ENA's doesn't.